src/app/accounts/index.ts

```diff
@@ -24,3 +25,26 @@ export const getBusinessMapMarkers = async () => {
   const accounts = AccountsRepository()
   return accounts.listBusinessesForMap()
 }
+
+export const toWalletIds = async (
+  account: Account,
+  walletNames: WalletName[],
+): Promise<WalletId[] | ApplicationError> => {
+  const wallets = WalletsRepository()
+
+  const walletIds: WalletId[] = []
+
+  for (const walletName of walletNames) {
+    const wallet = await wallets.findByWalletName(walletName)
+    if (wallet instanceof Error) {
+      return wallet
+    }
+
+    if (!account.walletIds.includes(wallet.id)) {
+      return new RepositoryError()
```

Comment:
Repository error is an error that only should be used at repository level. Please use a custom error, AuthorizationError, ValidationError

Reply:
Done